ext/CUDASupportExt.jl

```diff
@@ -0,0 +1,26 @@
+module CUDASupportExt
+using CUDA 
+using Adapt
+using ShiftedArrays
+using Base # to allow displaying such arrays without causing the single indexing CUDA error
+
+Adapt.adapt_structure(to, x::CircShiftedArray{T, D, CT}) where {T,D,CT<:CuArray} = CircShiftedArray(adapt(to, parent(x)), shifts(x));
+function Base.Broadcast.BroadcastStyle(::Type{T})  where (T<: CircShiftedArray{<:Any,<:Any,<:CuArray})
+    CUDA.CuArrayStyle{ndims(T)}()
+end
```

Comment:
```suggestion
parent_type(::Type{CircShiftedArray{T, N, S})  where {T, N, S} = S
Base.Broadcast.BroadcastStyle(::Type{T})  where (T<:CircShiftedArray} = Base.Broadcast.BroadcastStyle(parent_type(T))
```

I'm not sure if this is correct, but maybe something like the above is easier (same for ShiftedArrays)? This way everything is independent on CUDA and should work for all gpu array types.